.github/workflows/run_test.yml

```
@@ -0,0 +1,611 @@
+name: run-test
+
+on:
+  pull_request:
+    branches:
+      - main
+
+jobs:
+  run-test:
+    strategy:
+      fail-fast: false
+
+    runs-on: ubuntu-latest
+
+    steps:
+      - name: Set up go
+        uses: actions/setup-go@v5
+        with:
+          go-version: 1.22
```

to enabling go module cache
```suggestion
          go-version: 1.22
          cache: true
```